### DISCUSSION TITLE
Getting the paths of sharing files

### DISCUSSION THREAD

**Discussion by gorod0k** [Waboodoo/HTTP-Shortcuts#342] [Q&A] [answered]
Hello.
 I use this program to implement such a necessary, but such a difficult option in android, like passing a list of files to a script through the "Share" option.
 I was able to send a list of file names and file contents to the server, but there are 2 problems:
 1) I did not find an opportunity to also pass the initial path of the files in the system, which is very necessary.
 2) in addition to the paths, the files themselves are also transferred.  With a large number of files, this leads to an unjustified expenditure of operating system resources. 
 If necessary, I will send you a script and screenshots of the settings.
 I will be glad to your answer, with respect!

**Comment by Waboodoo** [comment, marked as answer]
Hello,
(First of all, sorry that I wasn't able to reply to your email yet, but I suppose answering here is easier now anyway.)

To address your issues:
1. This is currently not possible, and will likely remain so. Android's vision is going more and more in the opposite direction actually, where the concept of a file system and file paths are abstracted away (mostly for reasons of privacy and interoperability) and therefore not available to the apps that interact with the files. In fact, the data that is shared with the app might not even be a real file, it might be a virtual document, coming from cloud storage or generated on-the-fly. Because of this, there's no (reliable enough) way for the app to know the file's original path in the file system.

2. There's a workaround you could try here. You could have your shortcut read all the meta data from the files that you need, store them into variables, then trigger another shortcut which would read from those variables and make the actual HTTP request. The original shortcut could then abort itself. You can achieve this via the Scripting feature, with code that could look something like this:

```js
// Read data from the shared file, e.g. the file's name
const fileName = selectedFiles[0].name;

// Store the data into a variable (make sure to set the variable up first)
setVariable('my_file_name', fileName);

// Schedule the execution of another shortcut, which will make the HTTP request
enqueueShortcut('my_other_shortcut');

// Stop the execution of the current shortcut to prevent it from uploading the files
abort();
```

In your second shortcut, you could then use the variable's value, e.g. as part of the HTTP request body or in a script.

You'll find more information here:
- Variables: https://http-shortcuts.rmy.ch/variables
- Scripting: https://http-shortcuts.rmy.ch/scripting

Hope this helps somewhat.

**Comment by gorod0k** [comment]
Hello, thank you for the detailed answer, I admire the quality of your software product and documentation for it! Enjoy 👍
